### PR TITLE
vksc: use a different regkey for vksc driver lookup

### PR DESCRIPTION
**Pull request by zangold**
This change edits the PNP driver registry key name when compiling with -DVULKANSC=ON, to allow for Vulkan and VulkanSC drivers to coexist on the same system.

### CODE REVIEW

File: loader/vk_loader_platform.h
Comment (dgkoch) [inline loader/vk_loader_platform.h:450]: Alternatively we could modify the registry keys on lines 194-197 to put VulkanSC in the path instead of the actual key name (it might be less intrusive overall - as there are currently some tests that are failing with this change). 

And similarly for the keys below, otherwise.

Comment (aqnuep) [reply loader/vk_loader_platform.h:450]: Yes, that probably should be preferred.

Comment (dgkoch) [reply loader/vk_loader_platform.h:450]: Looking at https://github.com/KhronosGroup/VulkanSC-Loader/blob/sc_main/docs/LoaderLayerInterface.md#windows-layer-discovery it looks like both sets of keys are needed. 

I've suggested that in the case where we modify the PNP registry keys.. it would be preferable to modify how the Key is selected, so you don't need to ifdef each of the 6 different cases

eg if you could do something like "Vulkan" VARIANT "DriverNameWow"
and then VARIANT was defined as "" in the Vulkan case, and "sc" in the VulkanSC case

Comment (zangold) [reply loader/vk_loader_platform.h:452]: Made these changes in the most recent commit. Thanks for reviewing!

File: tests/framework/shim/windows_shim.cpp
Comment (dgkoch) [inline tests/framework/shim/windows_shim.cpp:40]: I'm not sure about this change - I presume you are doing this to be able to re-use the reg-key definitions?

From the perspective of having the test be able to detect unexpected registry key changes, it would probably be better to just repeat the logic to build up the correct Vulkan/VulkanSC regkey in the test file itself. 

@aqnuep not sure what you think from the maintainability perspective.

File: loader/vk_loader_platform.h
Comment (aqnuep) [reply loader/vk_loader_platform.h:452]: > eg if you could do something like "Vulkan" VARIANT "DriverNameWow"

Makes sense, but only if this change will actually get upstreamed. But the name `API_VARIANT` seems to generic and would prefer at least a `VK` prefix or something to avoid potential name collisions.

Also, defining the `_W` version for the widechar support should really be limited to Windows platforms.

File: tests/framework/shim/windows_shim.cpp
Comment (aqnuep) [reply tests/framework/shim/windows_shim.cpp:40]: Yes, I agree that this way we lose coverage for accidental changes.

**Review by dgkoch — CHANGES_REQUESTED** [review]

File: loader/vk_loader_platform.h
Comment (dgkoch) [reply loader/vk_loader_platform.h:452]: We should also update the LoaderLayerInterface.md document to specify the correct keys for VKSC.

**Review by dgkoch — APPROVED** [review]
LGTM

**Comment by aqnuep** [comment]
This looks fine to me, but we will need to upstream the changes (without the Vulkan SC specific parts) to https://gitlab.khronos.org/vulkan/Vulkan-Loader.

Considering that we did not yet upstream the loader API parameterization changes, I can include that in our effort taking care of that, so I'm fine with not having to block this on the upstreaming criteria.

**Comment by aqnuep** [comment]
@zangold, is this ready to be merged from your perspective?

**Comment by dgkoch** [comment]
@aqnuep - I believe so (I think he's out until the new year).

**Comment by aqnuep** [comment]
It seems like this PR ended up with a conflict with the `sc_main` branch after doing some clang-format related cleanup.

Could you please rebase and resolve any issues before we can merge it? I apologize for the inconvenience.

**Comment by aqnuep** [comment]
Never mind, it seems github showed incorrect status when I opened this.

**Review by aqnuep — APPROVED** [review]

**Pull request merged**
